File: backend/src/main/java/com/sustainablefarm/service/OperatorService.java
package com.sustainablefarm.service;

import com.sustainablefarm.model.Operator;
import com.sustainablefarm.model.Operator.ActiveStatus;
import com.sustainablefarm.model.Operator.Role;

import java.util.List;

/**
 * Service Interface for Operator Entity
 * Personnel data - Supporting Entity
 * 
 * @author Abdoul Ben Fatao SANON
 * @version 1.0.0
 */
public interface OperatorService {

    /**
     * Create new operator
     */
    Operator createOperator(Operator operator);

    /**
     * Update operator
     */
    Operator updateOperator(String operatorId, Operator operator);

    /**
     * Delete operator
     */
    void deleteOperator(String operatorId);

    /**
     * Get operator by ID
     */
    Operator getOperatorById(String operatorId);

    /**
     * Get all operators
     */
    List<Operator> getAllOperators();

    /**
     * Get operators by role
     */
    List<Operator> getOperatorsByRole(Role role);

    /**
     * Get operators by active status
     */
    List<Operator> getOperatorsByStatus(ActiveStatus status);

    /**
     * Get active operators by role
     * Business Rule: Only ACTIVE operators can be assigned
     */
    List<Operator> getActiveOperatorsByRole(Role role);

    /**
     * Get active QC inspectors
     */
    List<Operator> getActiveQcInspectors();

    /**
     * Get active auditors
     */
    List<Operator> getActiveAuditors();

    /**
     * Check if operator is active
     */
    boolean isOperatorActive(String operatorId);

    /**
     * Check if operator has specific role
     */
    boolean hasRole(String operatorId, Role role);

    /**
     * Update operator active status
     */
    Operator updateActiveStatus(String operatorId, ActiveStatus status);

    /**
     * Add certification to operator
     */
    Operator addCertification(String operatorId, String certification);

    /**
     * Get operators by certification
     */
    List<Operator> getOperatorsByCertification(String certification);
}

File: frontend/src/services/operatorService.js
import apiClient from './api';

const operatorApi = {
  getAll: () => apiClient.get('/api/operators'),
  getById: (id) => apiClient.get(`/api/operators/${id}`),
  create: (data) => apiClient.post('/api/operators', data),
  update: (id, data) => apiClient.put(`/api/operators/${id}`, data),
  delete: (id) => apiClient.delete(`/api/operators/${id}`),
  getByRole: (role) => apiClient.get(`/api/operators/role/${role}`),
  getByStatus: (activeStatus) => apiClient.get(`/api/operators/status/${activeStatus}`),
  getActive: () => apiClient.get('/api/operators/active'),
  getCertifiedByRole: (role) => apiClient.get(`/api/operators/certified/${role}`)
};

export { operatorApi };